earthmover/runs_file.py

```diff
@@ -158,8 +158,11 @@ def _build_hashes(self) -> Dict[str, str]:
             if f"$sources.{source.name}" not in node_data.keys():
                 continue
 
-            if not source.is_remote and source.file and not os.path.isdir(source.file):
-                sources_hash += self._get_file_hash(source.file)
+            if source.is_hashable and hasattr(source, 'file') and not os.path.isdir(source.file):
```

Comment:
`hasattr()` is needed here because a SqlSource has no `file` attribute, so simply `and source.file` throws an error.